addok/helpers/text.py

```diff
@@ -139,14 +141,25 @@ def __str__(self):
         return self._raw
 
 
-def compare_ngrams(left, right, N=2, pad_len=0):
+@lru_cache(maxsize=512)
+def ngrams(text, n=3):
+    # Make sure strings are at least 3 chars long, and a given token will have
+    # same ngrams whether at the start, the middle or the end of the string.
+    text = ' ' + text + ' '
+    return set([text[i:i+n] for i in range(0, len(text)-(n-1))])
```

Comment:
I think a reuse of the standard lib would be faster, something like that:

```python
from itertools import zip_longest
return zip_longest(*[iter(text)]*n)
```
See https://docs.python.org/3.4/library/itertools.html#itertools.zip_longest
and https://stackoverflow.com/questions/312443/how-do-you-split-a-list-into-evenly-sized-chunks

Reply:
zip_longest does not seem to allow to get similar result.

timeit also gives a ratio of  50000:1 in favor of actual code